ci-operator/step-registry/ibmcloud/deprovision/byok/ibmcloud-deprovision-byok-commands.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    rg=$1
+    export region
+    echo "Try to login to ${rg}..."
+    "${IBMCLOUD_CLI}" login -r ${region} -g ${rg} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+function run_command_with_retries() {
+  cmd="$1"
+  retries="$2"
+  interval="$3"
+
+  if [ X"$retries" == X"" ]; then
+      retries=20
+  fi
+
+  if [ X"$interval" == X"" ]; then
+      interval=30
+  fi
+
+  set +o errexit
+  output=$(eval "$cmd"); ret=$?
+  try=1
+
+  # avoid exit with "del Resource groups with active or pending reclamation instances can't be deleted"
+  while [ X"$ret" != X"0" ] && [ $try -lt $retries ]; do
+      sleep $interval
+      output=$(eval "$cmd"); ret=$?
+      try=$(expr $try + 1)
+  done
+  set -o errexit
+
+  if [ X"$try" == X"$retries" ]; then
+      return 2
+  fi
+  echo "$output"
+  return 0
+}
+
+cat ${SHARED_DIR}/metadata.json
+RESOURCE_GROUP=$(jq -r .ibmcloud.resourceGroupName ${SHARED_DIR}/metadata.json)
```

Comment:
Here we should NOT reply on `metadata.json`, it is created by installer when installing get started, but the kms key stuff are provisoned by users, we should save the resource group info when it is being provisioned, and get it from there when deprovision.

Reply:
How about get the resource group from ibm_kpKey.yaml

> platform:
>   ibmcloud:
>     defaultMachinePlatform:
>       bootVolume:
>         encryptionKey: "crn:v1:bluemix:public:kms:jp-tok:a/fdc2e14cf8bc4d53a67f972dc2e2c861:11772310-0ef1-497f-8517-1fab7820941e:key:1fdc67c9-2bee-45b0-8c1c-49c7ee3dc613"
>     resourceGroupName: ci-op-rdf29bp7-28a13-rg

Reply:
When provision, the order is:
1. provision resource group
2. provision kms instance
3. provision kms key
4. creating ibm_kpKey.yaml

once any step (e.g: 2, 3), you would never get `ibm_kpKey.yaml`, right? Then you would never can clean the created resource group.